internal/backend/runtime/omni/controllers/omni/internal/task/image/pulltask.go

```diff
@@ -77,25 +77,31 @@ func (p PullTaskSpec) RunTask(ctx context.Context, _ *zap.Logger, pullStatusCh P
 		currentNode = imageList.Node
 
 		for _, img := range imageList.Images {
+			currentError = nil
 			currentImage = img
-			currentNum++
-
-			if nodeImageInfoMap.shouldPull(imageList.Node, img) {
-				currentError = p.pullImage(ctx, clusterID, imageList.Node, img)
-				if currentError != nil {
-					errs = multierror.Append(errs, currentError)
-				}
-
-				if !channel.SendWithContext(ctx, pullStatusCh, PullStatus{
-					Request:    p.request,
-					Node:       currentNode,
-					Image:      currentImage,
-					CurrentNum: currentNum,
-					TotalNum:   totalNum,
-					Error:      currentError,
-				}) {
-					return errs
-				}
+
+			if !nodeImageInfoMap.shouldPull(imageList.Node, img) {
+				currentNum++
+
+				continue
+			}
+
+			currentError = p.pullImage(ctx, clusterID, imageList.Node, img)
```

Comment:
I wonder - does it retry on failure? or it simply fails and dies?

Reply:
retries, it stops only if you return a nil error